web-client/public/js/setuphandlers.js

```diff
@@ -0,0 +1,28 @@
+/* eslint no-unused-vars: [2, {"varsIgnorePattern": "updateApp|grnState"}] */
+var updateApp = require("./updateapp");
+export var setupHandlers = function (grnState) {
+
+    var GREY_EDGES_DASHED_MENU = "#grey-edges-dashed-menu";
+    var GREY_EDGES_DASHED_SIDEBAR = "#dashedGrayLineButton";
+
+    if (!$(GREY_EDGES_DASHED_MENU).prop("checked")) {
+        grnState.dashedLine = false;
+    } else {
+        grnState.dashedLine = true;
+    }
+
+    if ($(GREY_EDGES_DASHED_SIDEBAR).prop("checked")) {
+        grnState.dashedLine = true;
+    } else {
+        grnState.dashedLine = false;
+    }
+
+    $(GREY_EDGES_DASHED_MENU).click(function () {
+        updateApp.updateGreyEdgesAsDashedOptions(grnState.dashedLine);
+    });
+
+    $(GREY_EDGES_DASHED_SIDEBAR).on("change", function () {
+        updateApp.updateGreyEdgesAsDashedOptions(grnState.dashedLine);
+    });
```

Comment:
This is more of the idea, but still needs some tweaks. _It should update `grnState`_. Remember: MVC, with C looping back to M. So, when the UI elements change, that change updates `grnState` _inside the event handler_.

And remember, `updateApp` is _the_ update function. `updateApp.updateGreyEdgesAsDashedOptions` is too limited. It should just be `updateApp(grnState)`. That way, the code can factor in the _entire_ `grnState` object. You will need that for properties that affect each other.

So, these functions should look more like this:
```javascript
    grnState.dashedLIne = $(GREY_EDGES_DASHED_MENU).prop("checked");
    updateApp(grnState);
```
See how that works? MVC. When the user clicks `$(GREY_EDGES_DASHED_MENU)` (C), the first line updates `grnState`'s corresponding value to match (M). Then `updateApp` cycles back to make sure that the entire app is in sync with that change in `grnState` (V).

If you want to think of this in React terms, these functions are like the `onChange` functions in React. Note that the `onChange` functions (C) then call `setState` (M). When `setState` is called, it triggers a `render` call (V), which returns the new markup for the component. It's the same cycle, conceptually.

If you're concerned that this will lead to a lot of `change` and `click` functions that look the same, then you're right. But that is not the issue at hand. We can consolidate duplicated code later. For now, we want to establish the right structure. Again, this is _no different_ from the React cycle of having a state (`grnState`), having `onXXXXX` handlers (`click`, `change`, and friends), and having those handlers call `setState` (updating `grnState`). If there is anything that React does somewhat invisibly, it is that `setState` automatically calls `render`. In our case, all of our handlers will call `updateApp` explicitly.